ethcore/sync/src/block_sync.rs

```diff
@@ -231,6 +231,19 @@ impl BlockDownloader {
 		self.state = State::Blocks;
 	}
 
+	/// Reset sync to the specified block
+	pub fn reset_to_block(&mut self, start_hash: &H256, start_number: BlockNumber) {
+		self.reset();
+		self.last_imported_block = start_number;
+		self.last_imported_hash = start_hash.clone();
+		self.last_round_start = start_number;
+		self.last_round_start_hash = start_hash.clone();
+		self.imported_this_round = None;
+		self.round_parents = VecDeque::new();
+		self.target_hash = None;
+		self.retract_step = 1;
```

Comment:
What about `self.blocks`? It seems that `reset_to` does reset them too.

Reply:
It's being reset inside self.reset() call.